server/routes/card.js

```diff
@@ -0,0 +1,16 @@
+const express = require("express");
+const router = express.Router();
+const protect = require("../middleware/auth");
+const {
+  getCards,
+  createCard,
+  getCard,
+  updateCard,
+} = require("../controllers/card");
+
+router.route("/").get(protect, getCards);
+router.route("/").post(protect, createCard);
+router.route("/:id").get(protect, getCard);
+router.route("/:id").patch(protect, updateCard);
```

Comment:
Since all these routes have to use protect. A simpler solution would be something like

`
router.use(protect);

router.route("/").get(getCards);
router.route("/").post(createCard);
router.route("/:id").get(getCard);
router.route("/:id").patch(updateCard);
`